### PR TITLE
Spanish Latinamerican Sort added to keymap_extras 

### PR DESCRIPTION
**Pull request by shingonati0n**
## Description

Since I'm from Chile, I needed a layout akin to the layouts used locally. I found keymaps_spanish.h, however that keymap is for EU Spanish layout and not Latin American ones. Here I created the keymap for latin america, commonly known as Spanish Latinamerican Sort layout.  

- Keymap matches to keypresses on layout used in latin america
- Prefix for KC is SP_ (to avoid confusion with ES_ from keymap_spanish)

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
* core change so has to go via `develop`
* `keyboards/crkbd/rev1/rev1.c` needs to be reverted
* all submodules also need to be reverted
* `docs/reference_keymap_extras.md` also needs to be updated

**Comment by shingonati0n** [comment]
- I see you already moved to develop, thanks!
- reverted unwanted changes (rev1 and submodules)
- docs/reference_keymap_extras.md has been updated, please take a look

File: quantum/keymap_extras/keymap_spanish_latam.h [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "keymap.h"
+
+// clang-format off
+//Spanish Latinamerican sort
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_spanish_latam.h:22]:
```suggestion

```

File: quantum/keymap_extras/keymap_spanish_latam.h [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+ */
+// Row 1
+#define SP_NOT  ALGR(SP_PIPE) // ¬
+#define SP_BSLS ALGR(SP_QUOT) // (backslash)
+// Row 2
+#define SP_AT   ALGR(SP_Q)    // @
+#define SP_TILD ALGR(SP_PLUS) // ~
+// Row 3
+#define SP_CIRC ALGR(SP_LCBR) // ^ (dead)
+#define SP_GRV  ALGR(SP_RCBR) // ` (dead)
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_spanish_latam.h:149]:
```suggestion
#define SP_GRV  ALGR(SP_RCBR) // ` (dead)

```

File: quantum/keymap_extras/sendstring_spanish_latam.h [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    SP_H,    SP_I,    SP_J,    SP_K,    SP_L,    SP_M,    SP_N,    SP_O,
+    // p     q        r        s        t        u        v        w
+    SP_P,    SP_Q,    SP_R,    SP_S,    SP_T,    SP_U,    SP_V,    SP_W,
+    // x     y        z        {        |        }        ~        DEL
+    SP_X,    SP_Y,    SP_Z,    SP_LCBR, SP_PIPE, SP_RCBR, SP_PLUS, KC_DEL
+};
+
+
+
+
```
Comment (fauxpark) [inline quantum/keymap_extras/sendstring_spanish_latam.h:104]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Prefix should be `ES`. These should follow the ISO-3166 2-character country codes; though Latin America is obviously not a single country.

**Comment by shingonati0n** [comment]
- changed prefixing to adhere to ISO-3166 standard
- removed empty and unrelated lines to keep consistent format. Please take a look, thanks

File: quantum/keymap_extras/sendstring_spanish_latam.h [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    ES_X,    ES_Y,    ES_Z,    ES_LCBR, ES_QUOT, ES_RCBR, ES_LCBR, ES_MINS,
+    // `     a        b        c        d        e        f        g
+    ES_RCBR, ES_A,    ES_B,    ES_C,    ES_D,    ES_E,    ES_F,    ES_G,
+    // h     i        j        k        l        m        n        o
+    ES_H,    ES_I,    ES_J,    ES_K,    ES_L,    ES_M,    ES_N,    ES_O,
+    // p     q        r        s        t        u        v        w
+    ES_P,    ES_Q,    ES_R,    ES_S,    ES_T,    ES_U,    ES_V,    ES_W,
+    // x     y        z        {        |        }        ~        DEL
+    ES_X,    ES_Y,    ES_Z,    ES_LCBR, ES_PIPE, ES_RCBR, ES_PLUS, KC_DEL
+};
```
Comment (fauxpark) [inline quantum/keymap_extras/sendstring_spanish_latam.h:98]: There should be a single empty line at the end of this file.

**Review by fauxpark — APPROVED** [review]

File: quantum/keymap_extras/sendstring_spanish_latam.h (hunk 4 above)
Comment (shingonati0n) [reply quantum/keymap_extras/sendstring_spanish_latam.h]: Hi, just added as suggested, please take a look, thanks again!

**Comment by zvecr** [comment]
`quantum/keymap_extras/keymap_spanish_latam.h` now has to be implemented in <https://github.com/qmk/qmk_firmware/tree/develop/data/constants/keycodes/extras>. Please feel free to re-open the PR  with required changes.

**Pull request closed without merging**